testscript/src/org/partiql/testscript/junitRunner/PtsRunner.kt

```diff
@@ -43,35 +43,41 @@ class PtsRunner(private val testClass: Class<*>) : Runner() {
             .toList()
 
     private val ast = parser.parse(ionDocs)
-    private val testExpressions = compiler.compile(ast) 
-
-    override fun run(notifier: RunNotifier) = try {
-        val testResults = evaluator.evaluate(testExpressions)
-
-        testResults.forEach {
-            val testDescription = Description.createTestDescription(
-                    testClass, 
+    private val testExpressions = compiler.compile(ast)
+
+    override fun run(notifier: RunNotifier) {
+        testExpressions.forEach { testExp ->
+            val testResults = evaluator.evaluate(listOf(testExp))
```

Comment:
no reason to evaluate them 1 by 1, might as well run the evaluator for all `testExpressions`